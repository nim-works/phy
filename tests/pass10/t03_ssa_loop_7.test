discard """
  description: "
    Ensure liveness/reaching definitions analysis works with multiple nested
    loops.
  "
"""
(TypeDefs
  (Int 8)
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Locals (Type 0) (Type 0))
    (Stmts
      (Loop
        (Stmts
          (If (IntVal 1) (Break 1))
          (Loop
            (Stmts
              (If (IntVal 1) (Break 1))
              (Asgn (Local 0) (IntVal 200))
              (Drop (Copy (Local 1)))))
          (Loop
            (Stmts
              (If (IntVal 1) (Break 1))
              (Drop (Copy (Local 0)))
              (Asgn (Local 1) (IntVal 100))))))
      (Return))))
