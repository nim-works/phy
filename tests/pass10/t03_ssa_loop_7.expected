;$sexp
(TypeDefs
  (Int 8)
  (ProcTy (Void)))
(GlobalDefs)
(ProcDefs
  (ProcDef (Type 1)
    (Continuations
      (Continuation (Params)
        (Locals (Type 0) (Type 0))
        (Continue 1
          (List
            (Move (Local 0))
            (Move (Local 1)))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (SelectBool (IntVal 1)
          (Continue 3
            (List
              (Move (Local 0))
              (Move (Local 1))))
          (Continue 2 (List))))
      (Continuation (Params) (Locals)
        (Continue 14 (List)))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (Continue 4
          (List
            (Move (Local 0))
            (Move (Local 1)))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (SelectBool (IntVal 1)
          (Continue 6
            (List
              (Move (Local 1))))
          (Continue 5
            (List
              (Move (Local 0))
              (Move (Local 1))))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (Continue 8
          (List
            (Move (Local 0))
            (Move (Local 1)))))
      (Continuation
        (Params (Type 0))
        (Locals)
        (Continue 7 (IntVal 200)
          (List
            (Move (Local 0)))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (Drop
          (Copy (Local 1)))
        (Loop 4
          (List
            (Move (Local 0))
            (Move (Local 1)))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (Continue 9
          (List
            (Move (Local 0))
            (Move (Local 1)))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (SelectBool (IntVal 1)
          (Continue 11
            (List
              (Move (Local 0))))
          (Continue 10
            (List
              (Move (Local 0))
              (Move (Local 1))))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (Continue 13
          (List
            (Move (Local 0))
            (Move (Local 1)))))
      (Continuation
        (Params (Type 0))
        (Locals)
        (Drop
          (Copy (Local 0)))
        (Continue 12 (IntVal 100)
          (List
            (Move (Local 0)))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (Loop 9
          (List
            (Move (Local 1))
            (Move (Local 0)))))
      (Continuation
        (Params (Type 0) (Type 0))
        (Locals)
        (Loop 1
          (List
            (Move (Local 0))
            (Move (Local 1)))))
      (Continuation (Params) (Locals)
        (Continue 15 (List)))
      (Continuation (Params)))))
